\x
--- a/arch/arm64/boot/dts/qcom/ipq6018.dtsi	2024-03-28 14:14:27.431424230 +0800
+++ b/arch/arm64/boot/dts/qcom/ipq6018.dtsi	2024-03-28 14:14:01.366582092 +0800
@@ -215,6 +215,21 @@
 			no-map;
 		};
 
+		nss_region: memory@40000000 {
+			reg = <0x0 0x40000000 0x0 0x01000000>;
+			no-map;
+		};
+
+		tzapp_region: memory@49b00000 {
+		 	reg = <0x0 0x49b00000 0x0 0x00600000>;
+		 	no-map;
+		};
+
+		sbl: memory@4a500000 {
+			reg = <0x0 0x4a500000 0x0 0x00100000>;
+			no-map;
+		};
+
 		tz: memory@4a600000 {
 			reg = <0x0 0x4a600000 0x0 0x400000>;
 			no-map;
@@ -229,6 +244,16 @@
 			reg = <0x0 0x4ab00000 0x0 0x5500000>;
 			no-map;
 		};
+
+		q6_etr_region: memory@50000000 {
+			reg = <0x0 0x50000000 0x0 0x100000>;
+			no-map;
+		};
+
+		m3_dump_region: memory@50100000 {
+			reg = <0x0 0x50100000 0x0 0x100000>;
+			no-map;
+		};
 	};
 
 	smem {

