\x
--- a/arch/arm64/boot/dts/qcom/ipq6018.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq6018.dtsi
@@ -162,6 +162,26 @@
 			no-map;
 		};
 
+		nss_region: memory@40000000 {
+			reg = <0x0 0x40000000 0x0 0x01000000>;
+			no-map;
+		};
+
+		tzapp_region: memory@49b00000 {
+		 	reg = <0x0 0x49b00000 0x0 0x00600000>;
+		 	no-map;
+		};
+
+		uboot: memory@4a100000 {
+			reg = <0x0 0x4a100000 0x0 0x00400000>;
+			no-map;
+		};
+
+		sbl: memory@4a500000 {
+			reg = <0x0 0x4a500000 0x0 0x00100000>;
+			no-map;
+		};
+
 		tz: memory@4a600000 {
 			reg = <0x0 0x4a600000 0x0 0x00400000>;
 			no-map;
@@ -176,6 +196,16 @@
 			reg = <0x0 0x4ab00000 0x0 0x05500000>;
 			no-map;
 		};
+
+		q6_etr_region: memory@50000000 {
+			reg = <0x0 0x50000000 0x0 0x100000>;
+			no-map;
+		};
+
+		m3_dump_region: memory@50100000 {
+			reg = <0x0 0x50100000 0x0 0x100000>;
+			no-map;
+		};
 	};
 
 	rpm-glink {
