--- a/arch/arm64/Makefile	2024-03-25 19:34:08.989713592 +0800
+++ a/arch/arm64/Makefile	2024-03-25 19:33:52.584640195 +0800
@@ -20,12 +20,6 @@
 			$(call ld-option, --no-apply-dynamic-relocs)
 endif
 
-ifeq ($(CONFIG_ARM64_ERRATUM_843419),y)
-  ifeq ($(CONFIG_ARM64_LD_HAS_FIX_ERRATUM_843419),y)
-LDFLAGS_vmlinux	+= --fix-cortex-a53-843419
-  endif
-endif
-
 cc_has_k_constraint := $(call try-run,echo				\
 	'int main(void) {						\
 		asm volatile("and w0, w0, %w0" :: "K" (4294967295));	\


