--- a/Makefile	2024-03-25 19:34:08.989713592 +0800
+++ b/Makefile	2024-03-25 19:33:52.584640195 +0800
@@ -827,13 +827,32 @@
 KBUILD_CFLAGS	+= $(call cc-disable-warning, address-of-packed-member)
 
 ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE
-KBUILD_CFLAGS += -O2
+KBUILD_CFLAGS += -O3
 KBUILD_RUSTFLAGS += -Copt-level=2
 else ifdef CONFIG_CC_OPTIMIZE_FOR_SIZE
 KBUILD_CFLAGS += -Os
 KBUILD_RUSTFLAGS += -Copt-level=s
 endif
 
+# IPQ60XX 
+# CPU implementer	: 0x51
+# CPU variant	: 0xa
+# CPU part	: 0x801
+# Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
+# LLVM says 0x51, 0x801 so, optimize for cortex-a73?!
+# https://github.com/llvm/llvm-project/blob/main/llvm/lib/TargetParser/Host.cpp#L294
+KBUILD_CFLAGS += -pipe -mcpu=cortex-a73+crypto+aes+sha2 -mtune=cortex-a73 
+KBUILD_AFLAGS += -pipe -mcpu=cortex-a73+crypto+aes+sha2 -mtune=cortex-a73 
+# loop optimizations, but more like autism
+KBUILD_CFLAGS += -funswitch-loops -funroll-loops -fpeel-loops -fsplit-loops -fversion-loops-for-strides
+KBUILD_AFLAGS += -funswitch-loops -funroll-loops -fpeel-loops -fsplit-loops -fversion-loops-for-strides
+# GCC graphite, but this shit is unavailable on openwrt's gcc, enable it on gcc kbuild sometime
+# KBUILD_CFLAGS	+= -fgraphite-identity -floop-nest-optimize
+
+# force init stack all zero no matter what the config says
+KBUILD_CFLAGS	+= -ftrivial-auto-var-init=zero
+
+
 # Always set `debug-assertions` and `overflow-checks` because their default
 # depends on `opt-level` and `debug-assertions`, respectively.
 KBUILD_RUSTFLAGS += -Cdebug-assertions=$(if $(CONFIG_RUST_DEBUG_ASSERTIONS),y,n)

