--- a/Makefile
+++ b/Makefile
@@ -17,8 +17,8 @@ include $(PRJ_PATH)/Makefile.modules
 ####################################################################
 # 			SSDK-Style Makefile
 ####################################################################
-all: $(BIN_DIR) kslib
-	mkdir -p ./temp/;cd ./temp;cp ../build/bin/ssdk_ks_km.a ./;ar -x ssdk_ks_km.a; cp ../ko_Makefile ./Makefile;
+all: kslib $(BIN_DIR)
+	rm -rf ./temp/;mkdir -p ./temp/;cd ./temp;cp ../build/bin/ssdk_ks_km.a ./;ar -x ssdk_ks_km.a; cp ../ko_Makefile ./Makefile;
 	make -C $(SYS_PATH) M=$(PRJ_PATH)/temp/ CROSS_COMPILE=$(TOOLPREFIX) modules
 	cp $(PRJ_PATH)/temp/Module.symvers $(PRJ_PATH)/Module.symvers;
 	cp temp/*.ko build/bin;
@@ -28,7 +28,7 @@ all: $(BIN_DIR) kslib
 ####################################################################
 # 			LNX Modules-Style Makefile
 ####################################################################
-modules: $(BIN_DIR) kslib_c
+modules: kslib_c $(BIN_DIR)
 	@$(MAKE) -C $(SYS_PATH) M=$(PRJ_PATH) $(LNX_MAKEOPTS) modules
 	@echo "---Build [SSDK-$(VERSION)] at $(BUILD_DATE) finished."
 
--- a/make/target.mk
+++ b/make/target.mk
@@ -10,7 +10,9 @@ src_list_loop: src_list
 	$(foreach i, $(SUB_DIR), $(MAKE) -C $(i) src_list_loop || exit 1;)
 
 src_list:
-	echo -n "$(LOC_SRC_FILE) " >>  $(PRJ_PATH)/src_list.dep
+	echo "$(LOC_SRC_FILE) " >> $(PRJ_PATH)/src_list.dep
+	awk 'NF' $(PRJ_PATH)/src_list.dep > $(PRJ_PATH)/src_list.dep2
+	rm -rf $(PRJ_PATH)/src_list.dep; mv $(PRJ_PATH)/src_list.dep2 $(PRJ_PATH)/src_list.dep
 
 ####################################################################
 # 			SSDK-Style Makefile
